explore/future_compat.h

```diff
@@ -12,8 +12,13 @@
 
 #ifdef HAS_STD17
 #define ATTR(name) [[ name ]]
+// NOTE: c++ 17 detection defined above is not working properly for msvc.
```

Comment:
I recently made changes that should fix this - can you please check and revert this if it is fixed?